napytau/ingest/dataset_factory/napatau_dataset_factory.py

```diff
@@ -0,0 +1,146 @@
+        for fit_row in fit_rows:
+            (
+                distance_index,
+                shifted_intensity,
+                unshifted_intensity,
+                feeding_shifted_intensity,
+                feeding_unshifted_intensity,
+            ) = NapatauDatasetFactory.parse_fit_row(fit_row)
+            datapoint = datapoints.get_datapoint_by_distance(distance_index)
+            datapoint.set_intensity(shifted_intensity, unshifted_intensity)
+            if (
+                feeding_shifted_intensity is not None
+                and feeding_unshifted_intensity is not None
+            ):
+                datapoint.set_feeding_intensity(
+                    feeding_shifted_intensity, feeding_unshifted_intensity
+                )
+
+        return datapoints
+
+    @staticmethod
+    def parse_distance_row(distance_row: str) -> ValueErrorPair[float]:
+        split_row = distance_row.split(" ")
+        if len(split_row) < 3:
+            raise ValueError(
+                f"Expected at least 3 values in distance row, but got {len(split_row)}"
+            )
+
+        # The first value (at index 0) is a label, however since we index by distance we can ignore it # noqa E501
+        distance = float(split_row[1])
+        distance_error = float(split_row[2])
+
+        return ValueErrorPair(distance, distance_error)
+
+    @staticmethod
+    def parse_calibration_row(
+        calibration_row: str,
+    ) -> Tuple[float, ValueErrorPair[float]]:
+        split_row = calibration_row.split(" ")
+        if len(split_row) < 3:
+            raise ValueError(
+                f"Expected at least 3 values in calibration row, but got {len(split_row)}"  # noqa E501
+            )
+
+        distance_index = float(split_row[0])
+        calibration = float(split_row[1])
+        calibration_error = float(split_row[2])
+
+        return distance_index, ValueErrorPair(calibration, calibration_error)
+
+    @staticmethod
+    def parse_fit_row(
+        fit_row: str,
+    ) -> Tuple[
+        float,
+        ValueErrorPair[float],
+        ValueErrorPair[float],
+        Optional[ValueErrorPair[float]],
+        Optional[ValueErrorPair[float]],
+    ]:
+        split_row = fit_row.split(" ")
+        if len(split_row) < 5:
+            raise ValueError(
+                f"Expected at least 5 values in fit row, but got {len(split_row)}"
+            )
+
+        distance_index = float(split_row[0])
+        shifted_intensity = float(split_row[1])
+        shifted_intensity_error = float(split_row[2])
+        unshifted_intensity = float(split_row[3])
+        unshifted_intensity_error = float(split_row[4])
+
+        if len(split_row) == 5:
+            return (
+                distance_index,
+                ValueErrorPair(shifted_intensity, shifted_intensity_error),
+                ValueErrorPair(unshifted_intensity, unshifted_intensity_error),
+                None,
+                None,
+            )
+
```

Comment:
Can fit_rows contain comments starting with '#'? If so len(split_row) might not be equal to 5, even though only five values are defined.

Reply:
The specification does not mention this ability, also i don't see why you would annotate your data with comments